package com.valuemomentum.training.testing.junitdemo;

import org.junit.Assert;

import junit.framework.Test;

public class calculatorTest {

	
	private calculator calc;
	
	public void setup() throws Exception
	{
		calc=new calculator();
	}
	public void teardown() throws Exception
	{
		calc=null;
	}
	public void testAdd() 
	{
		double result=calc.add(10.5,20.33);
				assertEquals(30.42,result,0);
	}
	private void assertEquals(double d, double result, int i) {
		// TODO Auto-generated method stub
		
	}
	@Test
	public void testSubtract() {
		double result = calc.subtract(100.5, 100);
		  assertEquals(0.5, result, 0);
	}

	@Test
	public void testMultiply() {
		 double result = calc.multiply(2.5, 100);
		  assertEquals(250, result, 0);
	}

	@Test
	public void testDivide() {
		double result = calc.divide(100, 10);
		  Assert.assertEquals(10, result, 0.001);
	}
	
	@Test(expected = ArithmeticException.class)
	 public void testDivideByZero() {
	  calc.divide(100.5, 0);
	 }

	public static void main(String[] args) {
		// TODO Auto-generated method stub

	}

}
